### ISSUE TITLE
Weld compilation fails when materializing more than 5 merger

### ISSUE THREAD

**Issue by hvanhovell** [weld-project/weld#202]
Weld compilation fails if you try call result on more than 5 mergers. 

For example compilation of the following weld program:
```
|p0: vec[f64], p1: vec[f64], p2: vec[f64]|
 let merged = for(zip(p0, p1, p2),
                  {merger[i64, +], merger[f64, +], merger[i64, +], merger[f64, +], merger[i64, +], merger[f64, +]},
                  |bs1: {merger[i64, +], merger[f64, +], merger[i64, +], merger[f64, +], merger[i64, +], merger[f64, +]}, i1: i64, ns1: {f64, f64, f64}|
                   {merge(bs1.$0, 1L), merge(bs1.$1, f64(ns1.$0)), merge(bs1.$2, 1L), merge(bs1.$3, f64(ns1.$1)), merge(bs1.$4, 1L), merge(bs1.$5, f64(ns1.$2))});
 {result(merged.$0), result(merged.$1), result(merged.$2), result(merged.$3), result(merged.$4), result(merged.$5)}
```
...fails with the following error:
```
Compile error: module:1602:3: error: multiple definition of local value named 't.t200'
  %t.t200 = insertvalue %s1 undef, i64 %t.t199, 0
  ^
```
Using 5 or less mergers works.

**Comment by sppalkia** [comment]
Fixed in #203.

**Comment by hvanhovell** [comment]
Awesome! Thanks @sppalkia for the quick turnaround.
